File: data/code_pairs/fortran/1017.f90
module test
contains
  subroutine oacc1(result)
    implicit none
    integer :: i
    integer, intent(out) :: result
    integer, parameter :: n = 5
    integer :: a(n)
    result = 0

    !$acc data copyin(a) copyout(result)
    !$acc parallel loop reduction(+:result)
    do i = 1, n
        result = result + i
    end do
    !$acc end parallel loop
    !$acc end data
  end subroutine oacc1
end module test

File: data/code_pairs/cpp/1017.cpp
#include <iostream>
#include <openacc.h>

namespace test {
    void oacc1(int &result) {
        const int n = 5;
        int a[n];
        result = 0;

        #pragma acc data copyin(a) copyout(result)
        #pragma acc parallel loop reduction(+:result)
        for(int i = 1; i <= n; ++i) {
            result += i;
        }
    }
}